src/core/transform.js

```diff
@@ -14,30 +14,97 @@ var constants = require('./constants');
 
 /**
  * Multiplies the current matrix by the one specified through the parameters.
- * This is very slow because it will try to calculate the inverse of the
- * transform, so avoid it whenever possible.
+ * This is a powerful operation that can perform the equivalent of translate,
+ * scale, shear and rotate all at once.
  *
  * @method applyMatrix
- * @param  {Number} n00 numbers which define the 3x2 matrix to be multiplied
- * @param  {Number} n01 numbers which define the 3x2 matrix to be multiplied
- * @param  {Number} n02 numbers which define the 3x2 matrix to be multiplied
- * @param  {Number} n10 numbers which define the 3x2 matrix to be multiplied
- * @param  {Number} n11 numbers which define the 3x2 matrix to be multiplied
- * @param  {Number} n12 numbers which define the 3x2 matrix to be multiplied
+ * @param  {Number} n11 numbers which define the 2x3 matrix to be multiplied
+ * @param  {Number} n12 numbers which define the 2x3 matrix to be multiplied
+ * @param  {Number} n21 numbers which define the 2x3 matrix to be multiplied
+ * @param  {Number} n22 numbers which define the 2x3 matrix to be multiplied
+ * @param  {Number} n31 numbers which define the 2x3 matrix to be multiplied
+ * @param  {Number} n32 numbers which define the 2x3 matrix to be multiplied
```

Comment:
The original uses 0 index which is usually kore consistent with native APIs, do you think this can be changed to be 0 index as well?

Reply:
My reasoning behind this was the underlying web api uses notation that is closer to this: https://developer.mozilla.org/en-US/docs/Web/CSS/transform?v=a, 
in addition to most mathematical references using this notation.
Nethertheless, I'm certainly open to changing back to zero indexing - my main issue is that the original function arguments were incorrectly named for the zero-indexed case.

Reply:
I don't have a math or cs background so I couldn't really say which is better, I'm just more familiar with code stuff being zero-indexed. Let's have others pitch in what they think.

Reply:
the numbering does more closely align here, as counterintuitive as it may seem. however, I would suggest that we match the naming exactly as the mdn spec: https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/transform, or use a-f which may be less confusion to someone that doesn't understand what m11 vs dx mean.

Reply:
The base standard by W3C uses a-f: https://html.spec.whatwg.org/multipage/canvas.html#dom-context-2d-transform, so perhaps it's best to change to that? It also warns against using matrix numbering because the transposition is used in some applications where the numbers would be reversed. I don't know what would be best for 'beginners', though whether any beginner would actually use this function is questionable. Perhaps I should add a link to MDN/W3C to the documentation as well?

Reply:
yeah let's go with a-f. I think adding a link to one of those refs is a good idea, too.